projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -884,6 +884,88 @@ describe('DatagridComponent', () => {
                 expect(this.clrGridWidget.component.columns.first.customFilter).toEqual(true);
             });
         });
+
+        describe('@Input() columns', () => {
+            it('updates the columnsUpdatedObservable when set', function(this: HasFinderAndGrid): void {
+                this.finder.hostComponent.columns = [
+                    {
+                        displayName: 'Column',
+                        renderer: 'name',
+                    },
+                ];
+                this.finder.detectChanges();
+                this.finder.hostComponent.grid.columnsUpdatedObservable.subscribe(updated => {
+                    expect(updated).toBe(true);
+                });
+            });
+        });
+
+        describe('addColumn', () => {
+            beforeEach(function(this: HasFinderAndGrid): void {
+                this.finder.hostComponent.columns = [
+                    {
+                        displayName: 'Column',
+                        renderer: 'name',
+                    },
+                ];
+                this.finder.detectChanges();
+            });
+            it('adds the passed in column to list of existing grid columns', function(this: HasFinderAndGrid): void {
+                expect(this.finder.hostComponent.grid.columns.length).toBe(1);
+                this.finder.hostComponent.grid.addColumn({
+                    displayName: 'Column2',
+                    renderer: 'name2',
+                });
+                this.finder.detectChanges();
+                expect(this.finder.hostComponent.grid.columns.length).toBe(2);
+                expect(this.finder.hostComponent.grid.columns[1].displayName).toBe('Column2');
+            });
+            it(
+                'updates a existing column if a column exists with same display name as the column passed ' + 'in',
```

Comment:
Why are you concatenating this if it fits in one line?

Reply:
If I don't break the string, The Prettier Git hook moves the `function(this: HasFinderAndGrid): void {` above and that exceeds the maximum line length